Jenkinsfile

```diff
@@ -1,132 +1,325 @@
+
+            // return the job
+            if (combinationFilter(it)) {
+              return {
+                node(VECTORIZE_CODE) {
+                  def ARCH = (NODE_NAME == "KNL-Cluster-login") ? "KNL" : "HSW"
+                  results.put(it.join('-'), [:])
+                  results[it.join('-')].put("runOn", ARCH)
+                  stage("${it.join('-')}") {
+                    sh "rm -rf ${it.join('-')} || echo ''"
+                    ws("${WORKSPACE}/${it.join('-')}") {
+                      unstash 'repo'
+                      stage("build/${it.join('-')}") {
+                        try {
+                          printVariation(it)
+                          dir("src") {
+                            if(ARCH == "HSW") {
+                              sh "export VTK_INCDIR=/usr/include/vtk-6.3"
+                              sh "export VTK_LIBDIR=/usr/lib"
+                            } else {
+                              sh "source /etc/profile.d/modules.sh"
                             }
+                            sh """
+                              make TARGET=$TARGET \
+                                PARTYPE=$PARTYPE \
+                                PRECISION=$PRECISION \
+                                OPENMP=$OPENMP \
+                                REDUCED_MEMORY_MODE=$REDUCED_MEMORY_MODE \
+                                UNIT_TESTS=1 \
+                                VECTORIZE_CODE=$VECTORIZE_CODE \
+                                """ +
+                                ((ARCH=="KNL") ?
+                                  "VTK=0 CFG=" + (PARTYPE == "PAR" ? "icc-impi" : "icc") :
+                                  "VTK=1"
+                                ) + " -j4"
+                            sh "mv `readlink MarDyn` ${it.join('-')}"
+                            archiveArtifacts artifacts: "${it.join('-')}", onlyIfSuccessful: true
+                            stash includes: "${it.join('-')}", name: "${it.join('-')}"
+                            results[it.join('-')].put("build", "success")
+                          }
+                        } catch (err) {
+                          results[it.join('-')].put("build", "failure")
+                          error err
                         }
-                        stage('unit-test (knl)') {
-                            steps {
-                                build job: 'mardyn-unit-test-git-knl', parameters: [string(name: 'GIT_BRANCH', value: BRANCH_NAME)]
+                      }
+                      // FIXME: Mixed precision unit-tests fail with rmm
+                      if ((PRECISION=="MIXED").implies(REDUCED_MEMORY_MODE=="0")) {
+                        stage("unit-test/${it.join('-')}") {
+                          try {
+                            printVariation(it)
+                            if (ARCH=="HSW" && PARTYPE=="PAR") {
+                              sh "mpirun -n 4 ./src/${it.join('-')} -t -d ./test_input/"
```

Comment:
file should probably be unstashed beforehand? to ensure restarting of stages is possible

Reply:
ok, but then the stash is also unneeded, isn't it?

Reply:
It's only needed for the generator in the very end

Reply:
but only the repo stash, not the other one.

Reply:
Well, i used unarchive here, but i can move it to stash. For the mktcts generator, we need an executable: https://github.com/ls1mardyn/ls1-mardyn/pull/32/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R315

Reply:
still, stashing should be the way to do this, as we don't need the files afterwards. Can you make it, such that only that one file is stashed?

Reply:
Oh, sure, that's actually a good idea. The power of scripted pipelines!